Konscious.Security.Cryptography.Argon2/Argon2Lane.cs

```diff
@@ -18,18 +20,45 @@ public Argon2Lane(int blockCount)
                     throw new ArgumentOutOfRangeException(nameof(index));
                 }
 
-                return new Argon2Memory(_memory, 128 * index);
+                return _memory.Slice(128*index, 128);
             }
         }
 
-        public int BlockCount
+        public int BlockCount { get; }
+
+        private readonly Memory<ulong> _memory;
+    }
+
+    internal static class SpanExtensions
+    {
+        public static void Blit(this Span<ulong> toBlit, ReadOnlySpan<byte> bytes, int destOffset = 0, int srcOffset = 0, int byteLength = -1)
```

Comment:
This function runs reeeeeeaaaaaally fast now. It was fast before. Now it's "We have to run it 100m times to get a result that the benchmark doesn't complain about" fast